__tests__/gui/mapchip.feature

```diff
@@ -0,0 +1,44 @@
+Feature: Data Mapper
+
+  Scenario: Verify the data mapper is working correctly
+    Given I am on the Wazimap Homepage
+    Then I wait until map is ready
+
+    And I expand Data Mapper
+    Then Data Mapper should be displayed
+
+    And I click on "Demographics" in Data Mapper
+    And I select an indicator
+    Then I check if the choropleth filter dialog is collapsed
+    Then I check if choropleth legend is displayed
+    Then I check if description icon is visible
+    Then I check if tooltip is displayed on hover over description icon
+    Then I check filters applied label is visible
+    Then I check if tooltip is displayed on hover over applied filters label
+    Then I check applied filter label text
+    Then I check applied filters does not have notification badge
+
+    And I expand the choropleth filter dialog
+    Then I check filters applied label is hidden
+    Then I check if there is description
```

Comment:
The description should also be visible here right? As I understand .should('exist') will still pass if the element exists but is hidden

Reply:
hmm... I was not able to find a very good way of getting the length of text as element p is removed completely if there is no description.
Should i just remove this check as this can also be confirmed from a fixture or is there a better way to test this @emre2038 @michaelglenister ?

Reply:
I was just wondering if it would be better to set the description and test it with `.should('be.visible');`?

Reply:
@goyal1092 if p element is getting removed completely you can check with both `.should('exist')` and `.should('be.visible')` as Michael said. Because if the p element is there but hidden for some future bug, testing with just `.should('exist')` will give us a false positive

Reply:
cool 👍

Reply:
rather than that is there a way to check length of text inside <p> @emre2038 ?

Reply:
hmm.. I just want to have a test for icon only visible if there is description
I guess text comparison will work... though text is huge

Reply:
https://stackoverflow.com/questions/65669034/cypress-how-to-get-length-of-text